### PR TITLE
Add custom Sphinx extension that lists examples based on model name

### PR DESCRIPTION
**Pull request by jessica-mitchell**
This PR adds a custom Sphinx extension that displays a list of examples based on model name.

**How to use it**
Add directive to reStructured Text with required argument - the model_name:

`.. listexamples:: model_name`

This was used in `models/iaf_psc_alpha.h` and `aeif_cond_alpha`
See output [here](https://nest-simulator--2859.org.readthedocs.build/en/2859/models/iaf_psc_alpha.html#examples-using-this-model)

**Updated**

- Script now includes function that finds matches of model names in examples
- The titles of examples are displayed rather than filename
- Using the ref role syntax, which should allow the use of this directive anywhere in the documentation without issue (hopefully) 

**TODO**

- [x] Rename some example titles that are unclear
- [x] Fix sorting so it's used on the titles rather than filenames

### CODE REVIEW

File: doc/htmldoc/_ext/list_examples.py
Comment (pnbabu) [inline doc/htmldoc/_ext/list_examples.py:77]: Do we need to loop through the `sorted_examples`? Since it's a dictionary, we can directly get the value as `values = sorted_examples[key]`.

File: models/iaf_psc_alpha.h
Comment (pnbabu) [inline models/iaf_psc_alpha.h:156]: Should we also add this for all the other models?

**Review by pnbabu — CHANGES_REQUESTED** [review]
I think sorting the list of files by title text rather than the filenames is more useful. I see that it's listed as a TODO in this PR. Is that planned to be addressed here?

File: doc/htmldoc/_ext/list_examples.py
Comment (jessica-mitchell) [reply doc/htmldoc/_ext/list_examples.py]: You're right, Thanks for pointing that out!

File: models/iaf_psc_alpha.h
Comment (jessica-mitchell) [reply models/iaf_psc_alpha.h:156]: In principle yes, the main goal of this PR is to create the extension. I have only added it to 2 models, to see if the logic works and the correct output is displayed, but I can add it to all.

File: doc/htmldoc/_ext/list_examples.py
Comment (jessica-mitchell) [inline doc/htmldoc/_ext/list_examples.py:69]:
```suggestion
```

**Review by pnbabu — APPROVED** [review]
Looks good to me! This list is super helpful for someone browsing through the models. I think we could add the directive to all model files once you have approval from other reviewers as well.

File: models/iaf_psc_alpha.h
Comment (terhorstd) [reply models/iaf_psc_alpha.h:156]: Thanks, Jessica!

File: pynest/examples/spatial/conncon_targets.py
Comment (terhorstd) [inline pynest/examples/spatial/conncon_targets.py:24]:
```suggestion
Spatial networks: Convergent projection and rectangular mask, from source perspective
-------------------------------------------------------------------------------------
```

File: pynest/examples/spatial/connex_ew.py
Comment (terhorstd) [inline pynest/examples/spatial/connex_ew.py:24]:
```suggestion
Spatial networks: Circular mask and flat probability, with edge wrap
--------------------------------------------------------------------
```

**Review by terhorstd — APPROVED** [review]
Two bytes to cut, then I think it's ready to go.
Thanks for the cool upgrade! :+1:

**Pull request merged**
